src/person.h

```diff
@@ -263,14 +289,100 @@ class Mechanic : public Person {
 	void Assign(RideInstance *ri);
 	void NotifyRideDeletion(const RideInstance *ri);
 
-	bool DailyUpdate() override;
-	void DecideMoveDirection() override;
-	AnimateResult EdgeOfWorldOnAnimate() override;
 	AnimateResult VisitRideOnAnimate(RideInstance *ri, TileEdge exit_edge) override;
 	RideVisitDesire WantToVisit(const RideInstance *ri, const XYZPoint16 &ride_pos, TileEdge exit_edge) override;
 	void ActionAnimationCallback() override;
 
-	RideInstance *ride;  ///< The ride we're intending to inspect, if any.
+	const Money &GetSalary() const override
+	{
+		return SALARY;
+	}
+};
+
+/**
+ * A guard who walks around the park to stop guests from smashing up park property.
+ * @todo Vandalism is not implemented yet, so the guards don't actually have a function currently.
+ */
+class Guard : public StaffMember {
+public:
+	static const Money SALARY;
+
+	Guard();
+	~Guard();
+
+	void Load(Loader &ldr);
+	void Save(Saver &svr);
+
+	void ActionAnimationCallback() override
+	{
+		NOT_REACHED();  // Guards don't have action animations.
+	}
+
+	const Money &GetSalary() const override
+	{
+		return SALARY;
+	}
+};
+
+/**
+ * An entertainer who walks around the park to make guests more cheerful and patient.
+ * @todo Happiness and impatience are not implemented yet, so the entertainers don't actually have a function currently.
+ */
+class Entertainer : public StaffMember {
+public:
+	static const Money SALARY;
+
+	Entertainer();
+	~Entertainer();
+
+	void Load(Loader &ldr);
+	void Save(Saver &svr);
+
+	void ActionAnimationCallback() override
+	{
+		NOT_REACHED();  // Entertainers don't have action animations.
+	}
+
+	const Money &GetSalary() const override
+	{
+		return SALARY;
+	}
```

Comment:
I don't remember that the game allowed changing the salary, so it's a fixed number I guess.

An option here is to handle salary in the base class, as it's the same code for all types of staff. Simplest may be to have the salary constant be passed into the base class in its constructor.